nat-lab/tests/test_pmtu.py

```diff
@@ -0,0 +1,199 @@
+import config
+import pytest
+from contextlib import AsyncExitStack
+from helpers import SetupParameters, setup_environment, setup_connections
+from telio import AdapterType
+from telio_features import TelioFeatures
+from utils import testing, stun
+from utils.connection_util import (
+    ConnectionTag,
+    generate_connection_tracker_config,
+    ConnectionLimits,
+)
+from utils.ping import Ping
+from utils.router import IPStack
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                ip_stack=IPStack.IPv4,
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+            )
+        ),
+        # TODO(msz): Disable IPv6 tests since the docker netowrk uses local link addresses
+        #            causing `connect()` call to fail.
+        # pytest.param(
+        #     SetupParameters(
+        #         ip_stack=IPStack.IPv6,
+        #         connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+        #     )
+        # ),
+    ],
+)
+async def test_pmtu_back_hole(setup_params: SetupParameters) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await exit_stack.enter_async_context(
+            setup_environment(exit_stack, [setup_params])
+        )
+        client, *_ = env.clients
+
+        host = (
+            config.PMTU_PROBE_HOST_IP4
+            if setup_params.ip_stack in [IPStack.IPv4, IPStack.IPv4v6]
+            else config.PMTU_PROBE_HOST_IP6
+        )
+
+        await testing.wait_lengthy(client.probe_pmtu(host, 1300))
+
+        # do it second time bacause the kernel might cache something
+        await testing.wait_lengthy(client.probe_pmtu(host, 1300))
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                ip_stack=IPStack.IPv4,
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+            )
+        ),
+        # TODO(msz): Disable IPv6 tests since the docker netowrk uses local link addresses
+        #            causing `connect()` call to fail.
+        # pytest.param(
+        #     SetupParameters(
+        #         ip_stack=IPStack.IPv6,
+        #         connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+        #     )
+        # ),
+    ],
+)
+async def test_pmtu_with_nexthop(setup_params: SetupParameters) -> None:
```

Comment:
I wonder if it would make sense to wrap `inject-icmp-host-unreachable` into a docker container as well? It might be a bit more bloaty but then again - very easy to debug(you can just ssh into it and run tcpdump for example and inspect iptables) and also it would be more uniform with the existing setup

Reply:
Um, it is running inside the clients container

Reply:
Hmm... Shouldn't it run on the server's container instead?

Reply:
I mean it might work both ways, injecting packets directly on the client works but having it on the server would be a cleaner approach. A docker-container-per-case

Reply:
This script simply doesn't work on server because of iptables rules. I tried this approach but failed.